source/extensions/filters/network/sni_cluster/sni_cluster.cc

```diff
@@ -21,7 +21,8 @@ Network::FilterStatus SniClusterFilter::onNewConnection() {
     read_callbacks_->connection().streamInfo().filterState().setData(
         TcpProxy::PerConnectionCluster::key(),
         std::make_unique<TcpProxy::PerConnectionCluster>(sni),
-        StreamInfo::FilterState::StateType::Mutable);
+        StreamInfo::FilterState::StateType::Mutable,
+        StreamInfo::FilterState::LifeSpan::FilterChain);
```

Comment:
This one's hurting my brain.

I'd been reading the lifetimes as
    [L7]FilterChain,
    [L7]DownstreamRequest,
    [L4]DownstreamConnection,

This is a network object being added to the connection stream info.  The lifetime of thi sis [L4]Filterchain which is == to L4DownstreamConnection.  

I think we either need to rephrase the commentary since the way I read it, FilterChain is implicitly L7 filter chain.  Or we can change this to be DownstreamConnection because that's how long it will live and I think it'd be more clear that way.

Reply:
Ah, yes. I made this DownstreamConnection.